promotion/suggestions.go

```diff
@@ -166,8 +171,58 @@ func (service *Service) RedeemAndCreateSuggestionEvent(ctx context.Context, cred
 		return nil
 	}
 
-	service.eventChannel <- suggestion
-	// TODO emit event
+	// kafka event producer below - this could probably put into a generic module
+	kafkaBrokers := os.Getenv("KAFKA_BROKERS_STRING")
+	kafkaWriter := kafka.NewWriter(kafka.WriterConfig{
+		// by default we are waitng for acks from all nodes
+		Brokers:  []string{kafkaBrokers},
+		Topic:    "suggestion",
+		Balancer: &kafka.LeastBytes{},
+	})
+	defer closers.Panic(kafkaWriter)
+
+	// possible that this schema could be tightened more
+	// also, are all fields required?
+	// we re-read this file every time which is decidedly not fantastic
+	schema, err := ioutil.ReadFile("../schema-registry/grant/suggestion.avsc")
+	if err != nil {
+		return err
+	}
+
+	codec, err := goavro.NewCodec(string(schema))
+	if err != nil {
+		return err
+	}
+
+	// not thrilled with double encoding here; alternative is to find a nice way to turn this into a map
```

Comment:
maybe we could produce a map in `Suggest` above instead of this custom type

Reply:
Hey so just making sure - is this about producing a map in Suggest in order to avoid the intermediary json?  How would you like to get that map to RedeemAndCreateSuggestionEvent ? It seems like it wouldn't make sense to pass it to service.datastore.RunNextSuggestionJob but I am open.

Reply:
sorry, ignore my previous comment.

atm bytes are stored in the suggestion drain table that we use to buffer requested suggestions for validation and then emit the event. so in this function we should be dealing entirely with a serialized suggestion event.

so all of the avro encoding here should actually take place in `Suggest`. to avoid roundtripping from `SuggestionEvent` struct to native to binary bytes, I think we can just construct a map instead of a `SuggestionEvent` struct (that's the "native" representation in the avro lib we're used as I understand it). that way we can just do `BinaryFromNative`